app/code/Magento/InventorySales/Plugin/StockState/CheckQuoteItemQtyPlugin.php

```diff
@@ -0,0 +1,155 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventorySales\Plugin\StockState;
+
+use Magento\CatalogInventory\Api\Data\StockItemInterface;
+use Magento\CatalogInventory\Model\StockStateProvider;
```

Comment:
why do we have a dependency on particular implementation, but not an interface ? 
`Magento\CatalogInventory\Model\StockStateProvider`